msautotest/wxs/wfs_ogr_gpkg.map

```diff
@@ -95,6 +98,7 @@ LAYER
     "wfs_featureid"     "id"
     "gml_include_items" "all"
     "gml_types"         "auto"
+    "wfs_use_default_extent_for_getfeature" "no"
```

Comment:
is it really needed to reproduce the bug ? I'd anticipate that, be it set or not, the geometry passed to Intersects() would always be degenerate to the point

Reply:
When I don't set this, the extent is used to calculate the rect instead of the geometry in the intersect filter.

Reply:
I would expect them to be intersected at https://github.com/MapServer/MapServer/blob/main/mapogr.cpp#L2245

Reply:
My knowledge of MapServer is not extensive enough to completely understand the inner workings, but as I understand it, the rect is only set by the geometry from the filter when `msOGRExtractTopSpatialFilter` is called. This method sets `rect_is_defined` to true when calculating a `rect` was possible based on the filter. It looks like `msOGRExtractTopSpatialFilter` when `wfs_use_default_extent_for_getfeature` is `on`.

Reply:
hum, ok, anyway your fix is very reasonable.